### PR TITLE
adding support for file patterns

### PR DESCRIPTION
**Pull request by vsoch**
This PR will address #46, namely that it's currently not possible to define patterns of flies to match (e.g., dotfiles or similar). This PR will allow it to work as follows:

```bash
# Check only html files
urlchecker check --file-types *.html .

# Check hidden flies
# Note that since the shell would parse the *, we need to use quotes
urlchecker check --file-types ".*" .

# Check hidden files and html files
urlchecker check --file-types .*,*.html .
```

It works by way of using fnmatch, so technically any pattern glob you'd do on the command line should work! I also ran black for formatting, and updated the license dates. Once we update here, we can release and then update the action.

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

**Comment by vsoch** [comment]
@SuperKogito did we never set up CI? I don't see any tests running, and we don't have a GitHub action or anything similar! It looks like we used to have travis... is it no longer working?

**Comment by vsoch** [comment]
@SuperKogito I struggled with getting codecov to work here - I'm not sure if there is an envar missing or similar. I don't think it's a huge priority or rush, but when you have the time, could you take a look/

File: setup.py [hunk 1]
```diff
@@ -6,22 +6,23 @@
...
     with open(version_file) as fd:
         exec(fd.read(), lookup)
     return lookup
 
 
 # Read in requirements
-def get_reqs(lookup=None, key='INSTALL_REQUIRES'):
-    '''get requirements, mean reading in requirements and versions from
-    the lookup obtained with get_lookup'''
+def get_reqs(lookup=None, key="INSTALL_REQUIRES"):
```
Comment (vsoch) [inline setup.py:23]: @SuperKogito I think you have the codecov account? It should not be linting the setup.py for issues, we typically don't test that :)

**Comment by rootwork** [comment]
So, strangely, this works when there's one dotfile, but fails when there are multiple dotfiles.

Test:

1. `echo 'https://google.com' > .foo`
2. `urlchecker check --file-types .* .` [successfully tests google.com]
3. `echo 'https://google.com' > .bar`
4. `urlchecker check --file-types .* .` [Output: "Done. No urls were collected."]

File: setup.py (hunk 1 above)
Comment (vsoch) [reply setup.py:23]: okay, I've removed checking the setup.py, so we should be good! It looks like coverage went down a bit for one or two of the files - @SuperKogito let me know what tests you'd like to add (or if we should safely ignore).

**Comment by vsoch** [comment]
Oh strange, let me test that.

**Comment by rootwork** [comment]
And it looks like it's only a dotfiles thing -- I tried creating `foo.html` and `bar.html` with the same contents and it worked fine with one or both.

**Comment by vsoch** [comment]
Interesting, for me to get it to work (for one or both) I need a quote around the file types, like:

```bash
$ urlchecker check --file-types ".*" .
  original path: .
     final path: /tmp/test
      subfolder: None
         branch: master
        cleanup: False
     file types: ['.*']
          files: []
      print all: True
 url whitetlist: []
   url patterns: []
  file patterns: []
     force pass: False
    retry count: 2
           save: None
        timeout: 5

 /tmp/test/.foo 
 --------------
https://google.com

 /tmp/test/.bar 
 --------------
https://google.com


Done. All URLS passed.
```

**Comment by vsoch** [comment]
The reason is because argparse parses the non quoted one as just a single period (dot):
```
$ urlchecker check --file-types .* .
['.']
```

**Comment by vsoch** [comment]
And to go up one level, I believe the globbing is done by the shell (not the python client). So we perhaps just need to show using quotes, always.

**Comment by rootwork** [comment]
Yep, that makes sense, and works for me.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ and **Merged pull requests**. Critical items to know are:
 Referenced versions in headers are tagged on Github, in parentheses are for pypi.
 
 ## [vxx](https://github.com/urlstechie/urlschecker-python/tree/master) (master)
+ - adding support for dotfiles for a file type (0.0.21)
```
Comment (SuperKogito) [inline CHANGELOG.md:15]: maybe just say for dotfiles.
_for dotfiles for a file type_ is a bit confusing.

File: README.md [hunk 3]
```diff
@@ -100,9 +100,14 @@ check an entire directory. Let's clone and check the directory of one of the
...
+$ git clone https://github.com/SuperKogito/SuperKogito.github.io.git
+$ cd SuperKogito.github.io
+```
+
+and run the simplest command to check the present working directory (.).
+
+```bash
+$ urlchecker check .
 
 $ urlchecker check .
```
Comment (SuperKogito) [inline README.md:112]: I am not sure if this is intentional but `$ urlchecker check  .` is written twice.

File: README.md [hunk 4]
```diff
@@ -158,6 +163,24 @@ https://stackoverflow.com/questions/49197916/how-to-profile-cpu-usage-of-a-pytho
...
+urlchecker check --file-types *.html .
+
+# Check hidden flies
+urlchecker check --file-types ".*" .
+
+# Check hidden files and html files
+urlchecker check --file-types ".*,*.html" .
+```
+
+Note that while some patterns will work without quotes, it's recommended for most
```
Comment (SuperKogito) [inline README.md:180]: suggestion: can we make this important note bold or italic? 
what do you think?

**Review by SuperKogito — COMMENTED** [review]
Great work as usual @vsoch. I tested the feature locally and it works fine. 
I checked the code and it looks good, the tests are working and you did a great job removing unused code sections & imports. 

As for Travis CI, it is working fine [build/215832894](https://travis-ci.com/github/urlstechie/urlchecker-python/builds/215832894). The new GitHub CI has a nice formatting section but isn't the testing part redundant? 

As for the code coverage, I think you did enough work , so I won't ask for more. I also think that it is better if we write the tests for each others code to provide different perspective. I cannot tell when I will be able to work on it though :( So my suggestion is to ignore the coverage report for now :s 

Overall the feature looks great :clap:  :fire:

File: CHANGELOG.md (hunk 2 above)
Comment (vsoch) [reply CHANGELOG.md:15]: for the file type is in reference to the parameter. It's just the changelog, so I think more information is better.

File: README.md (hunk 3 above)
Comment (vsoch) [inline README.md:112]: Will fix!

```suggestion
```

File: README.md [hunk 5]
```diff
@@ -158,6 +162,24 @@ https://stackoverflow.com/questions/49197916/how-to-profile-cpu-usage-of-a-pytho
...
```
Comment (vsoch) [inline README.md:179]:
```suggestion
**Note that while some patterns will work without quotes, it's recommended for most**
```

File: README.md (hunk 4 above)
Comment (vsoch) [reply README.md]: Sure thing, done.

File: README.md (hunk 3 above)
Comment (vsoch) [reply README.md]: Second one is removed, that was definitely a mistake!

**Comment by vsoch** [comment]
Thanks @SuperKogito ! I figured that the redundant tests wouldn't hurt - if Travis ever goes away we can have it easily switch. Is it ok with you to leave as is, or would you like the github tests removed? Since they are different services they run at the same time, so it shouldn't slow anything down.

**Review by SuperKogito — APPROVED** [review]

**Comment by SuperKogito** [comment]
I just wanted to verify if they are redundant but I totally agree, I think the redundant tests wouldn't hurt, so let's keep them.

**Comment by vsoch** [comment]
Ok let's merge! I'll have some time this weekend to add the other fixes we talked about to the client/spelling, and then I'll draft a new release and update the action.

**Pull request merged**